### PR TITLE
fix(conditional formatting): controls looses on save

### LINKED ISSUE #21502: plugin-chart-table: Chart looses customization on save
When table chart is saved with conditional formatting, it reloads the page without those customizations.
But after reloading it again correctly sets the customizations.

#### How to reproduce the bug
1. create table chart on any data
2. in customize tab, add conditional formatting on any column
3. click on save
4. see colors disappear
5. reload the page again, should show colors in the cell correctly.

### Expected results

what you expected to happen.

### Actual results

what actually happens.

#### Screenshots

[Screencast from 17-09-22 03:07:57 PM IST.webm](https://user-images.githubusercontent.com/12967587/190850542-ab40f4d3-01f9-470f-8b34-6a90681131b7.webm)

### Environment

(please complete the following information):

- browser type and version: crome
- superset version:  latest master (last commit sha: 8c16806f5759ecc53ecef88a2e96e2e0964bffc6)
- python version: 3.9
- node.js version: 16.13.1
- any feature flags active: no

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by Always-prog**
### SUMMARY
Fixes #21502

## Before 

https://user-images.githubusercontent.com/66589759/220374482-e97429bc-f074-4ad8-81fd-8b5c3a0abea9.mp4

## After

https://user-images.githubusercontent.com/66589759/220374539-49bff0f6-6f75-41e6-82c7-ae67cac36840.mp4

### ADDITIONAL INFORMATION

- [X] Has associated issue: Fixes #21502
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by Always-prog** [comment]
@villebro Hello!
Can I get review for fix?

**Comment by villebro** [comment]
@Always-prog sorry for not getting to this sooner - are you able to rebase this PR?

**Comment by Always-prog** [comment]
@villebro Yes, I'll rebase it today

**Comment by Always-prog** [comment]
@villebro I rebased PR, can I get review?

**Review by villebro — COMMENTED** [review]
Tested to work as expected. However, I'd like to get sign-off from @kgabryje as I'm unsure if the previous `useCallback` is needed for perf reasons.

File: superset-frontend/src/explore/components/controls/ConditionalFormattingControl/ConditionalFormattingControl.tsx [hunk 1]
```diff
@@ -89,19 +84,15 @@ const ConditionalFormattingControl = ({
...
-    if (
-      newFormattingConfigs.length !== conditionalFormattingConfigs.length &&
-      onChange
-    ) {
-      setConditionalFormattingConfigs(newFormattingConfigs);
-      onChange(newFormattingConfigs);
-    }
-  }, [JSON.stringify(columnOptions)]);
-  useComponentDidUpdate(removeFormattersWhenColumnsChange);
+  const newFormattingConfigs = conditionalFormattingConfigs.filter(config =>
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/controls/ConditionalFormattingControl/ConditionalFormattingControl.tsx:87]: Can we wrap it in useEffect?

Comment (Always-prog) [reply superset-frontend/src/explore/components/controls/ConditionalFormattingControl/ConditionalFormattingControl.tsx]: Wrapped.
Can you please review my code again? 
Thank you!

**Review by kgabryje — APPROVED** [review]
Tested, works great. Thanks for fixing that!

**Pull request merged**